spec/datadog/tracing/contrib/sidekiq/server_internal_tracer/heartbeat_spec.rb

```diff
@@ -15,6 +15,8 @@
   end
 
   it 'traces the looping heartbeat' do
+    # TODO: Fix `expect_in_sidekiq_server`
+    skip('`expect_in_sidekiq_server` is giving false positive which is always green')
     expect_in_sidekiq_server do
       span = spans.find { |s| s.service == tracer.default_service && s.name == 'sidekiq.heartbeat' }
```

Comment:
I'm curious -- was this always incorrect, or does it just not work on modern versions of sidekiq?

Reply:
@ivoanjo Nice catch!

I did a more detailed investigation against different sidekiq versions. And found that the false positive results are only detected with  `6.4.1`, the version locked in our `contrib`, but `6.5.8` and `5.2.10` are working fine. 

I somehow noticed that the one spec was failing for `6.5.8` when I put a cap on [this](https://github.com/DataDog/dd-trace-rb/blob/master/Appraisals#L1129), but this seems to be a test setup issue, instead of an issue of our instrumentation code .

```
  2) Server internal tracer heartbeat traces the stop command
     Failure/Error: launcher = Sidekiq::Launcher.new(cli.send(:options))

     NoMethodError:
       undefined method `options' for #<Sidekiq::CLI:0x0000aaaace9c8dc0>
       Did you mean?  option_parser
```
Since this error occurs in the `run_sidekiq_server` method which prematurely exit the program. I believe we should be upgrading the supported version to `6.5.8` and then work on supporting `7.x`

PS. Sidekiq refactor internal on `6.5.0`. https://github.com/mperham/sidekiq/blob/main/Changes.md#650

Reply:
Just to clarify, do the changes in 6.5 affect only our tests or also our production support of sidekiq?

Reply:
I think it only affect our test suite